Datatable/AbstractDatatable.php

```diff
@@ -166,6 +167,10 @@ public function __construct(
 
         $this->authorizationChecker = $authorizationChecker;
         $this->securityToken = $securityToken;
+
+        if (!($translator instanceof LegacyTranslatorInterface) && !($translator instanceof TranslatorInterface)) {
```

Comment:
Are you sure this is required?
I suppose this should work with both `TranslatorInterface`.
No need to force people to use Contracts

Reply:
Yes, it makes sure that you're getting either the deprecated interface from the component (available in Symfony 4.4 and earlier) or the newer interface from the contract (available for Symfony 4.3 and newer).  The runtime check replaces the typehint that can't be enforced across multiple Symfony major versions.

Reply:
Oh yeah I didn't realize you added `!` on both conditions. I thought you wanted to force the use of Contracts.

But since the translation interface param is injected with DI, I am not sure this check is very useful.

Reply:
@Seb33300 the services.xml just passes in the @translator service ... so in 4.4 it'll be the previous class... in 5.0 it'll be the new contracts-interface.

Reply:
I know and this means this cannot be something else than a `TranslatorInterface`.

The the check is quite useless, except if the developer override the configuration to pass something else.

If you really want to keep this check I think this is better to place it at the beginning of the method.

Reply:
I don't think the contracts interface existed in 4.1 though; https://symfony.com/doc/4.1/translation.html